slave.mk

```diff
@@ -763,15 +763,24 @@ $(SONIC_INSTALL_DEBS) : $(DEBS_PATH)/%-install : .platform $$(addsuffix -install
 			{ while dpkg -s $(firstword $(subst _, ,$(basename $(deb)))) | grep "^Version: $(word 2, $(subst _, ,$(basename $(deb))))" &> /dev/null; do echo "waiting for $(deb) to be uninstalled" $(LOG); sleep 1; done } )
 		# put a lock here because dpkg does not allow installing packages in parallel
 		if mkdir $(DEBS_PATH)/dpkg_lock &> /dev/null; then
+
+		    # apt-get install inside sonic-slave may conflict with dpkg -i
+		    # add retry here
+		    for (( i=1; i<=10; i++))
```

Comment:
Can we reduce the complex logic of the two loops?
```
while true; do
   for (( i=1; i<=10; i++))
   done
done
```

Reply:
The first one is to acquire dpkg lock.
The second one is to for dpkg install. We can't reduce the two loops.

Reply:
If reduce one loop, we can remove dpkg lock.
Retry is enough.